tests/functional/test_show.py

```diff
@@ -79,14 +79,46 @@ def test_find_package_not_found():
     assert len(list(result)) == 0
 
 
+def test_report_single_not_found(script):
+    """
+    Test that a single unmatched package is reported not found.
+    """
+    # The following should report an error as there no result to print hence
+    # expect_error=True which implies allow_stderr_warning=True
+    result = script.pip('show', 'abcd-3', expect_error=True)
+    assert 'WARNING: Package(s) not found:' in result.stderr
```

Comment:
Is there a reason you can't assert equality for the entire warning line (perhaps after splitting)? It seems this would be simpler and more complete. But also, with the way the assertions are currently structured, it doesn't seem to be actually checking that the package name occurs in the warning message.

Reply:
Addressed